src/protocol.h

```diff
@@ -329,14 +329,11 @@ static inline bool MayHaveUsefulAddressDB(ServiceFlags services)
 class CAddress : public CService
 {
 public:
-    CAddress();
-    explicit CAddress(CService ipIn, ServiceFlags nServicesIn);
-
-    void Init();
+    CAddress() : CService{} {};
+    explicit CAddress(CService ipIn, ServiceFlags nServicesIn) : CService{ipIn}, nServices{nServicesIn} {};
 
     SERIALIZE_METHODS(CAddress, obj)
     {
-        SER_READ(obj, obj.Init());
```

Comment:
This may be fine here, but it's not obvious. When deserializing, the receiving object is expected to be overwritten completely. As `nTime` is only conditionally deserialized, the alternative would need a reset for that value.

Reply:
Thanks, fixed.